mesonpy/__init__.py

```diff
@@ -995,6 +996,34 @@ def _project(config_settings: Optional[Dict[Any, Any]]) -> Iterator[Project]:
         yield project
 
 
+def _get_ninja_if_needed() -> List[str]:
```

Comment:
This does not "get" ninja, it only checks if it is available. I would name this `_env_has_ninja_command()` or something, and then return a boolean instead of the dependency specification. I would probably also make sense to specify the required version as an argument: 
```python
def _env_has_ninja_command(version: str = _NINJA_REQUIRED_VERSION) -> bool:
    ...
```

Reply:
The current design was based on scikit-build-core, which may need to add multiple dependencies (cmake too, not just ninja), which is why it was designed this way. You are right, a flag is simpler with only one. (And technically might be simpler with two, might revisit there!)

Reply:
Haven't added the argument yet.